tool/gravity/cli/journal.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"compress/gzip"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/gravitational/gravity/lib/loc"
+	"github.com/gravitational/gravity/lib/localenv"
+	"github.com/gravitational/gravity/lib/state"
+	"github.com/gravitational/gravity/lib/system"
+	"github.com/gravitational/gravity/lib/system/mount"
+	"github.com/gravitational/gravity/lib/utils"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+func exportRuntimeJournal(env *localenv.LocalEnvironment, runtimePackage loc.Locator, outputFile string) error {
+	stateDir, err := state.GetStateDir()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	runtimePath, err := env.Packages.UnpackedPath(runtimePackage)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	rootDir := filepath.Join(runtimePath, "rootfs")
+	logDir := filepath.Join(state.LogDir(stateDir), "journal")
+	m := mount.NewMounter(rootDir)
+	if err := m.RoBindMount(logDir, journalDir); err != nil {
```

Comment:
What is this mount used for? Wouldn't just chrooting, doing export and then grabbing it from planet's rootfs work?

Reply:
After chroot, there's no way to access `<gravity-state-dir>/planet/log/journal` which lies outside of planet's rootfs, hence the need to mount it inside.

Reply:
Oh, it's cause /var/lib/gravity/planet is mounted inside planet. I see.